code/modules/explorer_drone/loot.dm

```diff
@@ -0,0 +1,187 @@
+	for(var/pack_type in unlockable_packs)
+		var/datum/supply_pack/P = SSshuttle.supply_packs[pack_type]
+		if(!P.special_enabled)
+			still_locked_packs += pack_type
+	if(!length(still_locked_packs)) // Just give out some cash instead.
+		var/datum/adventure_loot_generator/simple/cash/replacement = new
+		return replacement.generate()
+	var/chosen_pack_type = pick(still_locked_packs)
+	return new /obj/item/trade_chip(null,chosen_pack_type)
+
+/// Just picks and instatiates the path from the list
+/datum/adventure_loot_generator/simple
+	var/loot_list
+
+/datum/adventure_loot_generator/simple/generate()
+	var/loot_type = pick(loot_list)
+	return list(new loot_type())
+
+/// Unique exploration-only rewards - this is contextless
+/datum/adventure_loot_generator/simple/unique
+	id = "unique"
+	loot_list = list(/obj/item/clothing/glasses/geist_gazers,/obj/item/clothing/glasses/psych,/obj/item/firelance)
+
+/// Valuables
+/datum/adventure_loot_generator/simple/cash
+	id = "cash"
+	loot_list = list(/obj/item/storage/bag/money,/obj/item/antique,/obj/item/stack/spacecash/c1000,/obj/item/holochip/thousand)
+
+/// Drugs
+/datum/adventure_loot_generator/simple/drugs
+	id = "drugs"
+	loot_list = list(/obj/item/storage/pill_bottle/happy,/obj/item/storage/pill_bottle/lsd,/obj/item/storage/pill_bottle/penacid,/obj/item/storage/pill_bottle/stimulant)
+
+/// Rare minerals/materials
+/datum/adventure_loot_generator/simple/materials
+	id = "materials"
+	loot_list = list(/obj/item/stack/sheet/iron/fifty,/obj/item/stack/sheet/plasteel/twenty)
+
+/// Assorted weaponry
+/datum/adventure_loot_generator/simple/weapons
+	id = "weapons"
+	loot_list = list(/obj/item/gun/energy/laser,/obj/item/melee/baton/loaded)
+
+/// Pets and pet accesories in carriers
+/datum/adventure_loot_generator/pet
+	id = "pets"
+	var/carrier_type = /obj/item/pet_carrier/biopod
+	var/list/possible_pets = list(/mob/living/simple_animal/pet/cat/space,/mob/living/simple_animal/pet/dog/corgi,/mob/living/simple_animal/pet/penguin/baby,/mob/living/simple_animal/pet/dog/pug)
+
+/datum/adventure_loot_generator/pet/generate()
+	var/obj/item/pet_carrier/carrier = new carrier_type()
+	var/chosen_pet_type = pick(possible_pets)
+	var/mob/living/simple_animal/pet/pet = new chosen_pet_type()
+	carrier.add_occupant(pet)
+	return carrier
+
+/obj/item/antique
+	name = "antique"
+	desc = "Valuable and completly incomprehensible."
+	icon = 'icons/obj/exploration.dmi'
+	icon_state = "antique"
+
+/// Supply pack unlocker chip
+/obj/item/trade_chip
+	name = "trade contract chip"
+	desc = "Uses the station's cargo network to contact a black market supplier, allowing the purchase of a new crate type at cargo console."
+	icon = 'icons/obj/exploration.dmi'
+	icon_state = "trade_chip"
+	/// Supply pack type enabled by this chip
+	var/unlocked_pack_type
+
+/obj/item/trade_chip/Initialize(mapload, pack_type)
+	. = ..()
+	if(pack_type)
+		unlocked_pack_type = pack_type
+		var/datum/supply_pack/P = pack_type
```

Comment:
https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#use-descriptive-and-obvious-names